### PR TITLE
Must not include default reqwest ssl feature

### PR DESCRIPTION
**Pull request by ShayBox**
`reqwest` by default enables the `native-tls` feature, which includes `openssl-sys`, but some systems are not setup to compile this, and even though the user selects `rustls-tls` in their `Cargo.toml`, `vrchatapi-rust` still requires `native-tls` be compiled.

### CODE REVIEW

**Review by C0D3-M4513R — CHANGES_REQUESTED** [review]
These changes will disappear once a new version is generated.
Please edit the generator script.

Edit: 
Also counterpoint: if we were to accept this pr, people who don't select either `native-tls` or `rustls-tls` will not have any tls support.

Also I think this issue should be filed against the openapigenerator repo instead, as that is what we use to generate the rust repository.

**Comment by ShayBox** [comment]
Hmm, in that case passing the SSL features through this crate is the only option, it would work if the user manually depended on reqwest in their project, but not if they don't.

**Comment by ShayBox** [comment]
I don't know how to edit the generator to make it generate the appropriate changes, usually you only generate an OpenAPI library once as a base and modify it from there, I'll leave it to the generator maintainer to convert the pull request into a patch for the generator.

**Comment by jellejurre** [comment]
```
#!/usr/bin/env bash

# Generate Client
rm src/apis src/models docs -rf

./node_modules/\@openapitools/openapi-generator-cli/main.js generate \
-g rust \
--additional-properties=packageName=vrchatapi,supportAsync=true \
--git-user-id=vrchatapi \
--git-repo-id=vrchatapi-rust \
-o . \
-i https://raw.githubusercontent.com/vrchatapi/specification/gh-pages/openapi.yaml \
--http-user-agent="vrchatapi-rust"
#--global-property debugOperations=true

# Update entire description (replace entire line, match the random data there) line in Cargo.toml
sed -i 's/^description = ".*"/description = "VRChat API Client for Rust"/' Cargo.toml

# Remove messily pasted markdown at top of every file
find src -type f -exec sed -i '/VRChat API Banner/d' {} \;
# Remove openapi version in every file
find src -type f -exec sed -i '/The version of the OpenAPI document/d' {} \;
# Remove empty doc comments
find src -type f -exec sed -i '/^\s*\/\/\/\s*$/d' {} \;

# Cookie storage
sed -i 's/Client::new()/Client::builder().cookie_store(true).build().unwrap()/g' src/apis/configuration.rs
sed -i 's/features = \["json", "multipart"\]/default-features = false, features = \["json", "cookies", "multipart"\]/g' Cargo.toml

#Fix example
printf "\n[dev-dependencies]\ntokio = { version = '1', features = ['macros', 'rt-multi-thread'] }" >> Cargo.toml

# Add features to Cargo toml
printf "\n\n[features]\ndefault = [\"rustls-tls\"]\nnative-tls = [\"reqwest/native-tls\"]\nrustls-tls = [\"reqwest/rustls-tls\"]" >> Cargo.toml

# https://github.com/vrchatapi/specification/issues/241
cat patches/2FA_Current_User.rs >> src/models/current_user.rs
sed -i 's/pub use self::current_user::CurrentUser;/pub use self::current_user::{EitherUserOrTwoFactor, CurrentUser};/g' src/models/mod.rs
sed -i 's/Result<models::CurrentUser, Error<GetCurrentUserError>>/Result<models::EitherUserOrTwoFactor, Error<GetCurrentUserError>>/g' src/apis/authentication_api.rs

cargo build
cargo test
```

@ShayBox this generator.sh should add your changes to the generator

**Comment by C0D3-M4513R** [comment]
Okay, but just making a pr for those 2 features seems silly, when reqwest has a ton more.
These are all the tls related features as of the latest release:

[native-tls](https://docs.rs/crate/reqwest/latest/features#native-tls)
[native-tls-alpn](https://docs.rs/crate/reqwest/latest/features#native-tls-alpn)
[native-tls-vendored](https://docs.rs/crate/reqwest/latest/features#native-tls-vendored)
[rustls-tls](https://docs.rs/crate/reqwest/latest/features#rustls-tls)
[rustls-tls-manual-roots](https://docs.rs/crate/reqwest/latest/features#rustls-tls-manual-roots)
[rustls-tls-manual-roots-no-provider](https://docs.rs/crate/reqwest/latest/features#rustls-tls-manual-roots-no-provider)
[rustls-tls-native-roots](https://docs.rs/crate/reqwest/latest/features#rustls-tls-native-roots)
[rustls-tls-no-provider](https://docs.rs/crate/reqwest/latest/features#rustls-tls-no-provider)
[rustls-tls-webpki-roots](https://docs.rs/crate/reqwest/latest/features#rustls-tls-webpki-roots)

**Comment by C0D3-M4513R** [comment]
I don't see this as being useful, seeing as how [feature unification](https://doc.rust-lang.org/cargo/reference/features.html#feature-unification) exists.

(I'd expect that if I enabled any of the mentioned features above, the `default-tls` feature would be overridden. Though enabling multiple at the same time might not work. idk. I've not taken a look at reqwests source code.)

Edit:
Whilst that alone can apparently not be used to override the tls, I think we can use feature unification here to at least not have to add one feature for each reqwest tls variant.

**Pull request closed without merging**
